kin-core/src/main/java/kin/core/BackupRestoreImpl.java

```diff
@@ -41,10 +38,18 @@ public String exportWallet(@NonNull KeyPair keyPair, @NonNull String passphrase)
         return jsonify(keyPair.getAccountId(), salt, seed);
     }
 
+    private void initIfNeeded() {
+        if (!initialized) {
```

Comment:
could there be a threading issue? should this method also catch and throw an exception that can be handled by the UI instead of crashing?

Reply:
does calling NaCl.sodium(); twice from different thread - not very likely in any case

Reply:
yes, `KinClient` is not a thread safe.
you can catch import/export for `UnsatisfiedLinkError` at the upper level as some applicative handling, I don't see a reason to guard here against system error that native lib can't be loaded.